src/main/java/de/hhu/bsinfo/dxutils/UnsafeMemory.java

```diff
@@ -89,6 +89,26 @@ public static void copyBytes(final long p_address1, final long p_address2, final
         MS_UNSAFE_HANDLER.getUnsafe().copyMemory(null, p_address1, null, p_address2, p_length);
     }
 
+    /**
+     * Read data from memory into a boolean array.
+     *
+     * @param p_ptr
+     *         Start position in memory.
+     * @param p_array
+     *         Array to read the data into.
+     * @param p_arrayOffset
+     *         Start offset in array to start writing the booleans to.
+     * @param p_length
+     *         Number of booleans to read from specified start.
+     * @return Number of read elements.
+     */
+    public static int readBooleans(final long p_ptr, final boolean[] p_array, final int p_arrayOffset, final int p_length) {
+        MS_UNSAFE_HANDLER.getUnsafe().copyMemory(null, p_ptr, p_array,
+                UnsafeHandler.getArrayByteOffset() + p_arrayOffset, p_length);
```

Comment:
The base offset is not intended for boolean arrays. Please add a new method (`getArrayBooleanOffset`), which returns the appropriate value `sun.misc.Unsafe.ARRAY_BOOLEAN_BASE_OFFSET`.

Reply:
Done